test/langtools/jdk/javadoc/doclet/testNestedInlineTags/TestNestedLinkTag.java

```diff
@@ -0,0 +1,134 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug      8257925
+ * @summary  enable more support for nested inline tags
+ * @library  /tools/lib ../../lib
+ * @modules  jdk.javadoc/jdk.javadoc.internal.tool
+ * @build    javadoc.tester.*
+ * @run main TestNestedLinkTag
+ */
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Locale;
+
+import javadoc.tester.JavadocTester;
+import toolbox.ToolBox;
+
+public class TestNestedLinkTag extends JavadocTester {
+
+    public static void main(String... args) throws Exception {
+        TestNestedLinkTag tester = new TestNestedLinkTag();
+        tester.runTests(m -> new Object[] { Path.of(m.getName())});
+    }
+
+    ToolBox tb = new ToolBox();
+
+    enum TagKind {
+        LINK,
+        LINKPLAIN;
+        final String tag;
+        TagKind() {
+            tag = name().toLowerCase(Locale.ROOT);
+        }
+    }
+
+    enum DocLintKind {
+        DOCLINT("-Xdoclint"),
+        NO_DOCLINT("-Xdoclint:none");
+        final String option;
+        DocLintKind(String option) {
+            this.option = option;
+        }
+    }
+
+    @Test
+    public void testLinkLinkCombo(Path base) throws IOException {
+        for (TagKind outer : TagKind.values()) {
+            for (TagKind inner : TagKind.values()) {
+                for (DocLintKind dl : DocLintKind.values()) {
+                    Path dir = Files.createDirectories((base.resolve(outer + "-" + inner + "-" + dl)));
+                    testLinkLinkCombo(dir, outer, inner, dl);
+                }
+            }
+        }
+    }
+
+    void testLinkLinkCombo(Path base, TagKind outer, TagKind inner, DocLintKind dl) throws IOException {
+        Path src = base.resolve("src");
+        tb.writeJavaFiles(src,
+                """
+                    package p;
+                    /** First sentence. {@#OUTER# #m1() ABC {@#INNER# #m2() DEF} GHI} */
+                    public class C {
+                        /** Comment m1. */
+                        public void m1() { }
+                        /** Comment m2. */
+                        public void m2() { }
+                    }
+                    """
+                    .replaceAll("#OUTER#", outer.tag)
+                    .replaceAll("#INNER#", inner.tag));
+
+
+        javadoc("-d", base.resolve("api").toString(),
+                "-sourcepath", src.toString(),
+                dl.option,
+                "p");
+        checkExit(Exit.OK);
+
+        checkOutput("p/C.html", false,
+                "{@link",
+                "ABC <a href=\"#m2()\"");
+        checkOutput("p/C.html", true,
+                "ABC DEF GHI");
```

Comment:
Since you are running this with and without doclint, should this or any test in this file check for doclint warnings?